### DISCUSSION TITLE
Redirect Meilisearch logs to syslog

### DISCUSSION THREAD

**Discussion by curquiza** [meilisearch/product#543] [Feedback & Feature Proposal]
Following: https://github.com/meilisearch/meilisearch/issues/2811

Currently, Meilisearch returns logs on `stderr`. Some users seem to ask Meilisearch to return the information to [syslog](https://en.wikipedia.org/wiki/Syslog), the standard for message logging on unix systems.

A solution has been suggested in this PR: https://github.com/meilisearch/meilisearch/pull/2812

Related to:

https://github.com/meilisearch/product/discussions/471
https://github.com/meilisearch/product/discussions/504

**Comment by jirutka** [comment]


**Reply by curquiza** [reply]
Edited! thanks!

**Comment by jirutka** [comment]
FYI, the suggested PR meilisearch/meilisearch#2812 is already used in the [meilisearch](https://pkgs.alpinelinux.org/packages?name=meilisearch) package on Alpine Linux, to provide users with good integration to the system.

**Comment by mmachatschek** [comment]
Not having logs of background services (thats how meilisearch is running in the DO droplet) makes debugging very hard. We are facing some strange behaviors of our meilisearch DO droplet that just stops working out of nowhere.
Restarting the droplet/process makes everything work again. The process will stop working in the future at very uncertain times that we can't predict.

If there is a bug in meilisearch, we would be able to provide logs to debug this further if they would be saved to syslog or any other logfile.

This is also something that is on the roadmap https://roadmap.meilisearch.com/c/81-specify-log-path

IMO this would be beneficial to everyone in order to provide more useful debugging information especially in production environments.

cc @curquiza @gmourier

**Reply by mmachatschek** [reply]
friendly reminder @gmourier on your opinion (sorry for the ping)

**Reply by gmourier** [reply]
Hey @mmachatschek 👋

I'm sorry for my late answer, thanks for the ping! 

Just to make sure I understand, it is impossible to re-direct `stderr` output to a file on your end at the moment?

I agree that it could be useful to have the possibility to write the log on a user-chosen file path natively, today a simple workaround is to redirect those logs to a file when launching Meilisearch

Is the `stderr` format impracticable when redirected to a file?

**Reply by mmachatschek** [reply]
sorry @gmourier for missing your answer too 🙈 

I'm currently using the DO-marketplace template of meilisearch to host it. There are currently no redirects of the logs.

If our instance freezes and the DO host needs to be rebooted, we loose any information on the instance.

I would appreciate guidance on this topic if there is an easy way to do that for the default DO-marketplace meilisearch template.

**Reply by gmourier** [reply]
Hey @mmachatschek, just to be sure, you don't have access to any term to launch Meilisearch in your own way to redirect the log? e.g `./meilisearch --your-arguments-here 2>&1 | tee -a log`

**Comment by curquiza** [comment]
Not exactly a solution for this specific need, but in case people around here want to achieve in redirecting Meilisearch logs into a specific file: https://github.com/meilisearch/meilisearch/issues/898#issuecomment-1602975353
